File: ServiceModel/src/main/python/consumer.py
from zeep import Client

RAT = Client('/home/mohamed/Desktop/Paradigms-projects/Project-2/app/src/main/resources/RATService.wsdl').service

ans = True
while ans:
	print ("1. Get processes of the remote system\n2. Take a screenshot of the remote system and store it in your device\n3. Reboot the remote system\n4. Exit")
	ans=input("What would you like to execute?")
	if ans=="1": 
		result = RAT.getProcesses()
		print(result)
	elif ans=="2":
		screenshotName = input("Please give the screenshot's name: ")
		with open("screenshots/" + screenshotName + ".jpg", "wb") as binary_file:
			binary_file.write(RAT.takeScreenshot(screenshotName))
		print('Image Saved')
	elif ans=="3":
		waitingTime = input("Please give the waiting time before reboot: ")
		result = RAT.reboot(int(waitingTime))
		print(result)
	elif ans=="4":
		print("Exited successfully!")
		break
	elif ans !="":
		print("Not Valid Choice, please try again") 

File: AsynchronousClient/XMLSoap/src/main/js/consumer.js
let soap = require('soap-as-promised');
let fs = require('fs').promises;

let url = '/home/mohamed/Desktop/Paradigms-projects/Project-3/XMLSoap/app/src/main/resources/RATService.wsdl';

const clientPromise = soap.createClient(url);

const getProcessesPromise = clientPromise.then(RAT => RAT.getProcesses()).then(result => console.log(result.return)).catch(err => console.log(err));

let screenshotName = "screenshot1";

const takeScreenshotPromise = clientPromise.then(RAT => {
	return RAT.takeScreenshot({arg0: screenshotName});
}).then(result => {
	var buf = Buffer.from(result.return, 'base64');
	return fs.writeFile("screenshots/" + screenshotName + ".jpg", buf);
}).then(result => {
	console.log('Image Saved');
}).catch(err => {
	console.log(err);
});

let waitingTime = 180;

Promise.all([getProcessesPromise, takeScreenshotPromise]).then(values => clientPromise)
.then (RAT => RAT.reboot({arg0: waitingTime}))
.then(result => {
	console.log(result.return);
}).catch(err => {
	console.log(err);
});
